cw_nano/lib/nano_wallet.dart

```diff
@@ -85,22 +85,26 @@ abstract class NanoWalletBase
 
   // initialize the different forms of private / public key we'll need:
   Future<void> init() async {
+    if (_derivationType == DerivationType.unknown) {
+      _derivationType = DerivationType.nano;
+    }
     final String type = (_derivationType == DerivationType.nano) ? "standard" : "hd";
 
-    // our "mnemonic" is actually a seedkey:
+    // our "mnemonic" is actually a hex form seed:
     if (!_mnemonic.contains(' ')) {
-      _seedKey = _mnemonic;
+      _hexSeed = _mnemonic;
+      _mnemonic = "";
     }
 
-    if (_seedKey == null) {
+    if (_hexSeed == null) {
       if (_derivationType == DerivationType.nano) {
-        _seedKey = bip39.mnemonicToEntropy(_mnemonic).toUpperCase();
+        _hexSeed = bip39.mnemonicToEntropy(_mnemonic).toUpperCase();
       } else {
-        _seedKey = await NanoUtil.hdMnemonicListToSeed(_mnemonic.split(' '));
+        _hexSeed = await NanoUtil.hdMnemonicListToSeed(_mnemonic.split(' '));
```

Comment:
shouldn't this be the other way around?
you are checking if the derivation type is Nano, then you get the hex form from the mnemonic via bip39 ?!
and `else` (if the derivation type is bip39) you get the hex form through the NanoUtil?!!

Reply:
this is correct, the names for the derivation types are just not very good since the "nano standard" way of deriving keys still uses Bip39, it's just the derivation paths and signing algorithm are different

you can read more about it here: https://docs.nano.org/integration-guides/key-management/